### PR TITLE
Refactor/use currency amount for wallet store

### LINKED ISSUE #774: Use the CurrencyAmount data structure for the wallet redux store.
For now, the frontend uses BigNumber for the wallet redux store. This `BigNumber` is not fit the storing wallet's token balance info. Please check the logic here. https://github.com/balancednetwork/balanced-network-interface/blob/59c7cf0432a0153b95267dcf2c2a4515c5976923/src/store/wallet/reducer.ts#L10

After the tx routing PR, we have the `CurrencyAmount` data structure for balance info. Please use the `CurrencyAmount` instead of `BigNumber`. Also, use the address key instead of the symbol key. 

This data structure change will require a lot of code changes in the overall codebase. So please refactor the below items in sequence.

- [x] use the refactored hook `useCurrencyBalances` to fetch wallet balances in the `useWalletFetchBalances` hook. https://github.com/balancednetwork/balanced-network-interface/issues/772
- [x] refactor `WalletPanel`
- [x] refactor `SendPanel`
- [x] refactor `Unstake Panel` in sICX and ICX wallet
- [x] refactor `DepositPanel`
- [ ] refactor `CollateralPanel` and `LoanPanel`

### PR DESCRIPTION
**Pull request by lijac**
Close #774

### CODE REVIEW

File: src/store/wallet/hooks.ts [hunk 1]
```diff
@@ -2,71 +2,64 @@ import React, { useState, useEffect, useMemo } from 'react';
...
+      if (!JSBI.greaterThan(balance.quotient, BIGINT_ZERO)) return acc;
+
+      acc[balance.currency.wrapped.address] = balance;
+
+      return acc;
+    }, {});
+  }, [balances]);
+}
+
 export function useWalletFetchBalances(account?: string | null) {
```
Comment (octocat512) [inline src/store/wallet/hooks.ts:49]: After this refactoring, I thought that we can remove the wallet redux store. 
I think that we can `useWalletFetchBalances` hook and move the logic into the `useWalletBalances` hook.

**Review by octocat512 — CHANGES_REQUESTED** [review]

File: src/store/wallet/hooks.ts (hunk 1 above)
Comment (octocat512) [reply src/store/wallet/hooks.ts:49]: then we can remove this code.
https://github.com/balancednetwork/balanced-network-interface/blob/4a330afa77acfadc5590eb937f0f56e62da7a229/src/app/containers/TradePage/index.tsx#L22

**Review by octocat512 — CHANGES_REQUESTED** [review]

**Comment by lijac** [comment]
On Sejong network, swapping bnUSD for ICX fails. The error says:

InvalidParameter in updateBatchRewardsData

https://sejong.tracker.solidwallet.io/transaction/0x4ebe5eb007aa2f45353293222552c87e2136553e679d8a5db07faf4ae0ae425e

**Comment by lijac** [comment]
> On Sejong network, swapping bnUSD for ICX fails. The error says:
> 
> InvalidParameter in updateBatchRewardsData
> 
> https://sejong.tracker.solidwallet.io/transaction/0x4ebe5eb007aa2f45353293222552c87e2136553e679d8a5db07faf4ae0ae425e

@octocat512 @BennyOptions @gedanziger What would you suggest on this?

**Comment by octocat512** [comment]
@lijac we need to reduce using bignumber by using new data structures. please refactor the code again by removing bignumber usage.

**Comment by octocat512** [comment]
Please resolve conflicts. @lijac

**Pull request merged**